File: Java/timeDuendes.java
import java.util.ArrayList;
import java.util.Scanner;

class Duende{
    String nome;
    int idade;
    public Duende(){
        this.nome = "";
        this.idade = 0;
    }
    public void setNome(String s){
        this.nome = s;
    }
    public String getNome(){
        return this.nome;
    }
    public void setIdade(int x){
        this.idade = x;
    }
    public int getIdade(){
        return this.idade;
    }

    public String toString() {
        return nome + " " + idade;
    }
}

class timeDuendes{
public static void main(String[] args){
    Scanner sc = new Scanner(System.in);
    int n = 0;
    n = sc.nextInt();
    String nome = "";
    int idade = 0;
    ArrayList <Duende> time = new ArrayList<>();
    for(int i = 0; i < n; i++){
        Duende duende = new Duende();
        nome = sc.next();
        idade = sc.nextInt();
        duende.setNome(nome);
        duende.setIdade(idade);
        time.add(duende);
    }

    time.sort((b, a) -> {
        int compare = a.idade - b.idade;
        if (compare == 0) compare = a.nome.compareTo(b.nome);
        return compare;
    });

    ArrayList <Duende> lideres = new ArrayList<>();
    ArrayList <Duende> entregadores = new ArrayList<>();
    ArrayList <Duende> pilotos = new ArrayList<>();
    for(int i = 0; i < n/3; i++){
        lideres.add(time.get(i));
    }
    for(int j = 0; j < n/3; j++){
        entregadores.add(time.get(j + (n / 3)));
    }
    for(int k = 0; k < n/3; k++){
        pilotos.add(time.get(k + 2*(n/3)));
    }
    int count = 1;
    for(int i = 0; i < n/3; i++){
        System.out.println("Time " + count); count++;
        System.out.println(lideres.get(i));
        System.out.println(entregadores.get(i));
        System.out.println(pilotos.get(i));
        System.out.println();
    }
    
}
}

File: C/timeDuendes.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>


typedef struct Duende{
    char* nome;
    int idade;
}Duende;

int compareTo(const void *a, const void *b){
    Duende* a1 = (Duende*)a;
    Duende* b1 = (Duende*)b;
    int resp = b1->idade - a1->idade;
    if(resp == 0) resp = strcmp(a1->nome,b1->nome);

    return resp;
}


int main(void){
    int n = 0;
    scanf("%d", &n);
    Duende* time = (Duende*)malloc(n * sizeof(Duende));

    for(int i = 0; i < n; i++){
        Duende duende;
        duende.nome = (char*)calloc(50,sizeof(char));
        scanf("%s", duende.nome);
        scanf("%d", &duende.idade);
        time[i] = duende;
    }

    qsort(time, n, sizeof(Duende), compareTo);
    
    Duende* lideres = (Duende*)malloc(n * sizeof(Duende));
    Duende* entregadores = (Duende*)malloc(n * sizeof(Duende));
    Duende* pilotos = (Duende*)malloc(n * sizeof(Duende));

    for(int i = 0; i < n/3; i++){
        lideres[i] = time[i];
    }
    for(int j = 0; j < n/3; j++){
        entregadores[j] = time[j + n/3];
    }
    for(int k = 0; k < n/3; k++){
        pilotos[k] = time[k + 2 * (n/3)];
    }
    int count = 1;
    for(int i = 0; i < n/3; i++){
        printf("Time %d\n", count); count++;
        printf("%s %d\n", lideres[i].nome, lideres[i].idade);
        printf("%s %d\n", entregadores[i].nome, entregadores[i].idade);
        printf("%s %d\n", pilotos[i].nome, pilotos[i].idade);
        printf("\n");
    }

    return 0;
}